src/serializers/encode_context.c

```diff
@@ -19,10 +20,7 @@ inline void GraphEncodeContext_Reset(GraphEncodeContext *ctx) {
 	assert(ctx);
 	ctx->state = INIT;
 	ctx->keys_processed = 0;
-	ctx->processed_nodes = 0;
-	ctx->processed_edges = 0;
-	ctx->processed_deleted_nodes = 0;
-	ctx->processed_deleted_edges = 0;
+	ctx->processed_entities = 0;
 	ctx->datablock_iterator = NULL;
 	ctx->current_relation_matrix_id = 0;
 	ctx->matrix_tuple_iterator = NULL;
```

Comment:
Wouldn't you prefer `Reset` to free iterators in case they're not NULL ?

Reply:
I don't know in which scenario it could happen, but better safe than sorry. Added.